### DISCUSSION TITLE
Difference between sync and install?

### DISCUSSION THREAD

**Discussion by berislavlopac** [pdm-project/pdm#1913] [General]
I'm trying to understand what is (if any) the difference between `pdm sync` and `pdm install`? I was expecting `sync` to remove installed packages that are not in the lock file (e.g. if installed via `pip`), but apparently that is not the case.

**Comment by frostming** [comment]
They have some overlap, `install` even calls `sync` behind the scenes, but they both have some unique flags to do different jobs.

`sync` does remove unneeded packages if you pass `--clean` or `--only-keep` flag, which are not available for install command.

**Comment by stfdxv** [comment]
It's really surprising that `pdm sync` doesn't remove stray packages by default.
At the same time `pdm remove some_package_added_via_pdm` does remove stray packages.
